test/TestCaseReader.cpp

```
@@ -137,6 +146,56 @@ pair<SourceMap, size_t> TestCaseReader::parseSourcesAndSettingsWithLineNumber(is
 				if (sources.count(currentSourceName))
 					BOOST_THROW_EXCEPTION(runtime_error("Multiple definitions of test source \"" + currentSourceName + "\"."));
 			}
+			else if (boost::algorithm::starts_with(line, externalSourceDelimiterStart) && boost::algorithm::ends_with(line, sourceDelimiterEnd))
+			{
+				string externalSource = boost::trim_copy(line.substr(
+					externalSourceDelimiterStart.size(),
+					line.size() - sourceDelimiterEnd.size() - externalSourceDelimiterStart.size()
+				));
+				string externalSourceName = externalSource;
+
+				// Does the external source define a remapping?
+				size_t remappingPos = externalSource.find('=');
+				if (remappingPos != string::npos)
+				{
+					externalSourceName = boost::trim_copy(externalSource.substr(0, remappingPos));
+					externalSource = boost::trim_copy(externalSource.substr(remappingPos + 1));
+				}
+
+				fs::path testCaseParentDir = canonical(fs::path{m_fileName}).parent_path();
+				fs::path externalSourcePath(externalSource);
+				if (!fs::path(externalSource).is_relative())
+					BOOST_THROW_EXCEPTION(runtime_error(string("External Source need to be relative.")));
+				fs::path externalSourceFullPath = testCaseParentDir / externalSourcePath;
+
+				string externalSourceContent;
+				if (fs::exists(externalSourceFullPath))
+					externalSourceContent = util::readFileAsString(externalSourceFullPath.string());
+				else
+					BOOST_THROW_EXCEPTION(runtime_error("External Source '" + externalSourcePath.string() + "' not found."));
+
+				if (!externalSourceName.empty())
+				{
```

I think we should just report an error (using `BOOST_THROW_EXCEPTION`) when the name is empty. Or is there are valid situation where this is useful?

---

I put this as an assert.

---

This can actually happen in normal circumstances. I.e. if you just do `==== ExternalSources: ====` and forget the name. I think that a normal error is more appropriate in this case.